plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -354,43 +362,64 @@ open class KtlintPlugin : Plugin<Project> {
             group = VERIFICATION_GROUP
             description = "Runs ktlint on all kotlin sources in this project."
         }
+    }
 
     private fun Project.getAndroidVariantMetaKtlintCheckTask(
         variantName: String,
         multiplatformTargetName: String? = null
-    ): Task {
+    ): TaskProvider<Task> {
         val taskName = "ktlint${variantName.capitalize()}${multiplatformTargetName?.capitalize() ?: ""}Check"
-        return tasks.maybeCreate(taskName).apply {
-            group = VERIFICATION_GROUP
-            description = "Runs ktlint on all kotlin sources for android $variantName variant in this project."
+        return try {
+            tasks.named(taskName)
+        } catch (ignored: UnknownDomainObjectException) {
+            taskHelper(taskName) {
+                group = VERIFICATION_GROUP
+                description = "Runs ktlint on all kotlin sources for android $variantName variant in this project."
+            }
         }
     }
 
-    private fun Project.getMetaKtlintFormatTask(): Task = this.tasks.findByName(FORMAT_PARENT_TASK_NAME)
-        ?: this.task(FORMAT_PARENT_TASK_NAME).apply {
+    private fun Project.getMetaKtlintFormatTask(): TaskProvider<Task> = try {
+        this.tasks.named(FORMAT_PARENT_TASK_NAME)
+    } catch (ignored: UnknownDomainObjectException) {
+        taskHelper(FORMAT_PARENT_TASK_NAME) {
             group = FORMATTING_GROUP
             description = "Runs the ktlint formatter on all kotlin sources in this project."
         }
+    }
 
     private fun Project.getAndroidVariantMetaKtlintFormatTask(
         variantName: String,
         multiplatformTargetName: String? = null
-    ): Task {
+    ): TaskProvider<Task> {
         val taskName = "ktlint${variantName.capitalize()}${multiplatformTargetName?.capitalize() ?: ""}Format"
-        return tasks.maybeCreate(taskName).apply {
-            group = FORMATTING_GROUP
-            description = "Runs ktlint formatter on all kotlin sources for android $variantName" +
-                " variant in this project."
+        return try {
+            tasks.named(taskName)
+        } catch (ignored: UnknownDomainObjectException) {
+            taskHelper(taskName) {
+                group = FORMATTING_GROUP
+                description = "Runs ktlint formatter on all kotlin sources for android $variantName" +
+                    " variant in this project."
+            }
         }
     }
 
-    private fun setCheckTaskDependsOnKtlintCheckTask(project: Project, ktlintCheck: Task) {
-        project.tasks.findByName("check")?.dependsOn(ktlintCheck)
+    private fun setCheckTaskDependsOnKtlintCheckTask(
+        project: Project,
+        ktlintCheck: TaskProvider<KtlintCheckTask>
+    ) {
+        try {
+            project.tasks.named("check").configure { it.dependsOn(ktlintCheck) }
+        } catch (e: UnknownDomainObjectException) {
+            project.logger.info("Check task is not available.")
+        }
```

Comment:
@big-guy Is there a better way of doing this in Gradle 5.0?

@Tapchicoma I don't know if the logging is valuable here.

Reply:
Instead of doing this, only configure the `check` task when the `base` plugin is applied.

Reply:
It is not only about checking if `check` task is available or not. But also about not causing an early configuration. If you configure it without `named`, then you cause other configurations to happen too.